src/dialogs/more-info/components/lights/dialog-light-color-favorite.ts

```diff
@@ -0,0 +1,125 @@
+import { mdiClose } from "@mdi/js";
+import { css, CSSResultGroup, html, LitElement, nothing } from "lit";
+import { customElement, property, state } from "lit/decorators";
+import { fireEvent } from "../../../../common/dom/fire_event";
+import "../../../../components/ha-button";
+import "../../../../components/ha-dialog";
+import "../../../../components/ha-dialog-header";
+import { EntityRegistryEntry } from "../../../../data/entity_registry";
+import { LightColor } from "../../../../data/light";
+import { haStyleDialog } from "../../../../resources/styles";
+import { HomeAssistant } from "../../../../types";
+import "./light-color-picker";
+import { LightColorFavoriteDialogParams } from "./show-dialog-light-color-favorite";
+
+@customElement("dialog-light-color-favorite")
+class DialogLightColorFavorite extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @state() _dialogParams?: LightColorFavoriteDialogParams;
+
+  @state() _entry?: EntityRegistryEntry;
+
+  @state() _color?: LightColor;
+
+  public async showDialog(
+    dialogParams: LightColorFavoriteDialogParams
+  ): Promise<void> {
+    this._entry = dialogParams.entry;
+    this._dialogParams = dialogParams;
+    await this.updateComplete;
+  }
+
+  public closeDialog(): void {
+    if (this._dialogParams?.cancel) {
+      this._dialogParams.cancel();
+    }
+    this._dialogParams = undefined;
+    this._entry = undefined;
+    fireEvent(this, "dialog-closed", { dialog: this.localName });
+  }
+
+  private _colorChanged(ev: CustomEvent) {
+    this._color = ev.detail;
+  }
+
+  private async _save() {
+    if (!this._color) {
+      this.closeDialog();
+      return;
+    }
+
+    this._dialogParams?.submit?.(this._color);
+    this._dialogParams = undefined;
+    this._entry = undefined;
+    fireEvent(this, "dialog-closed", { dialog: this.localName });
```

Comment:
We should make a single function that does close handling and cleanup, otherwise we forget one of them when we make changes